docs/release_notes.rst

```diff
@@ -10,6 +10,7 @@ Current Development Version
 New Features
 
 * Skip function arguments prefixed with `_` in D417 check (#440).
+* Add flag to disable `# noqa` comment processing in API (#485).
```

Comment:
If we do plan on adding this to the checker - I am not sure if it makes sense to expose it only programmatically without exposing it via the CLI or the config file.

Reply:
Personally I agree with you, but I didn't implement the CLI or config options because of @sigmavirus24's comments: https://github.com/PyCQA/pydocstyle/issues/484#issuecomment-640879812 https://github.com/PyCQA/pydocstyle/issues/484#issuecomment-641263791

I'm happy to add CLI and/or config support. My preference would be to add it in a separate PR as there seems to be some debate about landing that, and this change has utility regardless. See #486

Getting this merged is blocking adding support for this to flake8-docstyle which is turn is blocking https://github.com/plinss/flake8-noqa/issues/1

Reply:
moved